argparse: Correct argparse tuple help docs

jchristgit opened the pull request:
The second argument must not be a string as-is. Add a testcase to
validate that it throws a validation error.

max-au commented:
>  Second element of the tuple can be either a string printed as-is, a

It's the documentation that's wrong here. There is no way to tell the list from the string. I am surprised there are no test cases that ensure the failure for this: `help => {<<"[-s SHARD]">>, [1,2,3]}}`

Hence the need to wrap the string in the extra list like `["string is nested"]`.

Would you mind changing the PR to actually state that gotcha, and a comment (or even a specific test case)?

jchristgit commented:
> It's the documentation that's wrong here.
> 
> Would you mind changing the PR to actually state that gotcha, and a comment (or even a specific test case)?

Thank you for the feedback. I've added the gotcha to the documentation
and added a test case. Let me know what you think.

max-au commented:
Looks good to me, now it'd be good to get OTP team opinion too. Maybe I just missed a way to have a simpler ("just string") variant (e.g. `binary()` type).

jhogberg commented:
I thought it was good enough as-is, so I've merged it. Thanks for the PR! :)
